### ISSUE TITLE
Support for Gzip Post Bodies

### ISSUE THREAD

**Issue by belfare** [logstash-plugins/logstash-input-http#33]
We need support for gzip in this plugin so that we can minimize the transit size of the request.  We can force this off right now, but gzip would be extremely useful.

**Comment by dwapstra** [comment]
I have used https://gist.github.com/mattbrictson/9106778 to update the plugin.
commit  https://github.com/dwapstra/logstash-input-http/commit/018550ddd3315f732bfecfde936ece2f5fe54ad2

**Comment by jsvd** [comment]
@dwapstra nice! can you do a PR with that? and a couple of tests? :pray:

**Comment by jsvd** [comment]
You can base off a test of gzip payload from [this test](https://github.com/logstash-plugins/logstash-input-http/blob/master/spec/inputs/http_spec.rb#L25-L38):


```ruby
  describe "#run" do
    subject { LogStash::Inputs::Http.new }
    before :each do
      subject.register
      Thread.new { subject.run(queue) }
    end
    it "should include remote host in \"host\" property" do
      agent.post!("http://localhost:8080/meh.json",
        :headers => { "content-type" => "text/plain" },
        :body => "hello")
      event = queue.pop
      expect(event["host"]).to eq("127.0.0.1")
    end
  end
```

But instead the body and headers will be adapted to a gziped or compressed request

**Comment by dwapstra** [comment]
I've created a pull request. Need to look into the tests

**Comment by dwapstra** [comment]
@jsvd  I have added test cases for the deflate and gzip requests.  The pass if correctly exeuted, however if I put non-compressed data the test run seems to hang...

**Comment by dwapstra** [comment]
I have added error handling and test cases for decompression errors

**Comment by jsvd** [comment]
implemented in #40 by @dwapstra, big thanks for the PR and tests

logstash-input-http 2.1.0 has been released with it: https://rubygems.org/gems/logstash-input-http/versions/2.1.0
